app/controllers/mixins/actions/vm_actions/ownership.rb

```diff
@@ -102,6 +102,14 @@ def build_ownership_info(ownership_ids)
           session[:edit] = @edit
         end
 
+        def resource_types_for_header(resource_table)
+          if @ownershipitems.present?
+            @ownershipitems.map { |x| ui_lookup(:model => model_for_vm(x).to_s) }.uniq.join(", ")
```

Comment:
While I understand the motivation behind this and I agree that it's quite valid, I don't think this is the right way to fix these kind of problems. In general, it's not a good idea to do any kind of string arithmetics with strings, that will be propagated into the UI and translated.

While the above fix may look good and work in English, it won't look good in other languages: for example, we cannot assume that other languages use the comma the same way English does.

Here's what I'd suggest.

We set ownership for images, instances, vms and templates, right? So we can have a routine, which will produce the whole header string like:
```ruby
if @ownershipitems.present?
  _('Set ownership for selected items')
else
  case resource_table
  when ...
    _('Set owenrship for virtual machine')
  when ...
    _('Set ownership for instance')
  when ...
  end
end
```
So no string concatenation, no `ui_lookup()` and the translated strings will actually make sense.

Reply:
@mzazrivec thanks for the review 👍